Fixed the panic in subquery pullup.

guofengrichard opened the pull request.

guofengrichard committed:
Fixed the panic in subquery pullup.

guofengrichard committed:
Revise on the sublink pullup.

danielgustafsson commented on src/backend/optimizer/plan/subselect.c:
```diff
@@ -1468,6 +1456,29 @@ convert_ANY_sublink_to_join(PlannerInfo *root, SubLink *sublink,
 		return NULL;
 
+	/*
+	 * GPDB_92_MERGE_FIXME:
```
Does this need a FIXME comment once merged? Are we leaving any known open items from this PR?

guofengrichard replied:
I believe there is no known issues from this PR, but not with full confidence. Will remove the FIXEME once this PR is reviewed.

danielgustafsson reviewed (commented):
Seems reasonable.

guofengrichard committed:
Fixes in convert_EXISTS_sublink_to_join.

Function 'convert_EXISTS_sublink_to_join' may be called
twice to pull up EXISTS sublink. We need to copy 'subselect'
so we can modify it safely.

danielgustafsson commented:
> On 28 Jun 2018, at 10:15, Richard Guo <notifications@github.com> wrote:
> 
> @guofengrichard commented on this pull request.
> 
> In src/backend/optimizer/plan/subselect.c <https://github.com/greenplum-db/gpdb-postgres-merge/pull/23#discussion_r198745893>:
> 
> > @@ -1468,6 +1456,29 @@ convert_ANY_sublink_to_join(PlannerInfo *root, SubLink *sublink,
>  	if (!bms_is_subset(upper_varnos, available_rels))
>  		return NULL;
>  
> +	/*
> +	 * GPDB_92_MERGE_FIXME:
> I believe there is no known issues from this PR, but not with full confidence. Will remove the FIXEME once this PR is reviewed.
> 
Figured as much, thanks for clarifying

guofengrichard committed:
Remove FIXME tag
